packages/front-end/components/Billing/PaymentInfo.tsx

```diff
@@ -0,0 +1,264 @@
+      throw new Error(e.message);
+    }
+  }
+
+  async function detachPaymentMethod(paymentMethodId: string) {
+    try {
+      const methodIndex = paymentMethods.findIndex(
+        (method) => method.id === paymentMethodId
+      );
+
+      if (paymentMethods.length === 1 && subscription?.status !== "canceled") {
+        throw new Error(
+          "Unable to delete payment method. You must have at least 1 payment method on file."
+        );
+      }
+
+      if (methodIndex <= -1) {
+        throw new Error(
+          "Cannot delete: Payment method does not exist on this subscription"
+        );
+      }
+      await apiCall("/subscription/payment-methods/detach", {
+        method: "POST",
+        body: JSON.stringify({
+          paymentMethodId,
+        }),
+      });
+
+      const updatedData = paymentMethods.toSpliced(methodIndex, 1);
+      setPaymentMethods(updatedData);
+    } catch (e) {
+      throw new Error(e.message);
+    }
+  }
+
+  useEffect(() => {
+    if (canShowPaymentInfo) {
+      fetchPaymentMethods();
+    }
+  }, [apiCall, canShowPaymentInfo, fetchPaymentMethods, subscription]);
+
+  if (!canShowPaymentInfo) return null;
+
+  return (
+    <>
+      {paymentMethodModal ? (
+        <StripeProvider>
+          <AddPaymentMethodModal
+            onClose={() => setPaymentMethodModal(false)}
+            refetch={() => fetchPaymentMethods()}
+            numOfMethods={paymentMethods.length}
+          />
+        </StripeProvider>
+      ) : null}
+      {defaultPaymentMethod ? (
+        <Modal
+          header="Update default payment method"
+          open={true}
+          cta="Set as default payment method"
+          submit={async () => await setPaymentMethodAsDefault()}
+          trackingEventModalType=""
+          close={() => setDefaultPaymentMethod(undefined)}
+        >
+          Are your sure? The default payment method will be the one charged on
+          future invoices.
+        </Modal>
+      ) : null}
+      <div className="p-3 border mb-3">
+        <Flex justify="between" align="center" className="pb-3">
+          <h3 className="mb-0">Payment Methods</h3>
+          <div>
+            <Tooltip
+              body="You can only have up to 3 payment methods on file"
+              shouldDisplay={paymentMethods.length > 2}
+            >
+              <Button
+                disabled={paymentMethods.length > 2}
+                onClick={() => {
+                  setPaymentMethodModal(true);
+                  track("Edit Payment Method Modal", {
+                    source: "payment-method-empty-state",
+                  });
+                }}
+              >
+                <span className="h4 pr-2 m-0 d-inline-block align-top">
+                  <GBAddCircle />
+                </span>
+                Add Payment Method
+              </Button>
+            </Tooltip>
+          </div>
+        </Flex>
+        {error ? (
+          <Callout status="warning">{error}</Callout>
```

Comment:
Nit: should we uses `status="error"` instead here?  Or is this usually not a real error and just a warning?